code/game/objects/effects/decals/turfdecal/tilecoloring.dm

```diff
@@ -60,6 +73,14 @@
 /obj/effect/turf_decal/trimline/Initialize()
 	if(SSevents.holidays && SSevents.holidays[APRIL_FOOLS])
 		color = "#[random_short_color()]"
+	if(SSevents.holidays && SSevents.holidays[PRIDE_WEEK])
+		for(var/obj/effect/turf_decal/tile/T in get_turf(src))
+			if(!T.pride_already_applied)
+				T.color = GLOB.pride_month_colors[GLOB.current_pride_color]
```

Comment:
You could make it so the rainbow always draws in the same color horizontally by doing this:
T.color = GLOB.pride_month_colors[(y % GLOB.pride_month_colors.len) + 1]
This may look odd in long horizontal hallways though, would need to see it ingame to really judge it.

Reply:
It would also remove the need to loop over all the tile decals on the turf.

Reply:
Will take a look at this.